### DISCUSSION TITLE
Zods schema to output file

### DISCUSSION THREAD

**Discussion by ppalmeida** [orval-labs/orval#900] [Ideas]
Should Orval output Zods schema for the Types as well?
Does anyone else think this could be useful?

**Comment by angusd3v** [comment]
I would find this useful -- one thing I do in my project that I'd rather not have to do manually is parse/validate query params from a URL that need to correspond to known enum values that I have orval codegenned types for (but not zod schemas that could help me do the parsing/validating). Just digging around the repo to see if there's anything existing I could leverage.

**Reply by ppalmeida** [reply]
Interesting use case. I do understand that it could create a coupling to Zod (or others like Yup or Joi). But having schemas to validate responses in runtime could be a great addition to the hooks exported by Orval.

**Comment by devraj** [comment]
Wouldn't this be solved by just doing something like:
```tsx
export async function performLogin(loginData: z.infer<typeof loginFormSchema>)
```

**Comment by vincerubinetti** [comment]
Definitely think this would be useful. As @devraj says, you can "solve" this by just using `z.infer`. But take my case as an example. I'm trying to replace a bunch of manually defined schemas with auto-generated ones. My manual schemas look like this, where I export both the runtime validator and the type side-by-side, with the same name for consistency:

```ts
export const ontology = z.object({
  id: z.string(),
  type: z.string(),
  name: z.string(),
  description: z.string(),
  series: z.string(),
});

export type Ontology = z.infer<typeof ontology>;
```

Orval replaces the first declaration there, but I still have to write out the second. Is it the biggest deal to still manually write the `z.infer`? No. But multiply the above code by 100, and it becomes more of a slog, and more of an opportunity for naming-mismatches, forgetting exports, and other errors.

It'd be great to have a bit more flexibility with how Orval exports things. Though as a package maintainer myself, I also respect that that can be a pandora's box.
